appliances/MinIO/appliance.sh

```diff
@@ -0,0 +1,331 @@
+  Refer to MinIO documentation for additional information.
+  - The appliance will use /dev/vdb as the first disk for MinIO storage, and continue with vdc, vdd, etc. If any disk has a XFS partition already created, 
+the configure script will use that partition to mount on the corresponding path.
+
+After the VM has booted and app
+EOF
+)
+ONE_SERVICE_RECONFIGURABLE=true
+
+### Contextualization defaults #######################################
+
+ONEAPP_MINIO_ROOT_USER="${ONEAPP_MINIO_ROOT_USER:-myminioadmin}"
+ONEAPP_MINIO_ROOT_PASSWORD="${ONEAPP_MINIO_ROOT_PASSWORD:-minio-secret-key-change-me}"
+ONEAPP_MINIO_OPTS="${ONEAPP_MINIO_OPTS:---console-address :9001}"
+
+### Globals ##########################################################
+
+###############################################################################
+###############################################################################
+###############################################################################
+
+#
+# service implementation
+#
+
+service_cleanup()
+{
+    :
+}
+
+service_install()
+{
+    # ensuring that the setup directory exists
+    #TODO: move to service
+    mkdir -p "$ONE_SERVICE_SETUP_DIR"
+
+    # minio install
+    install_minio_server
+    # create user and group for MinIO
+    create_user_group
+    # create MinIO environment variable file
+    create_default_environment_variable_file
+    # create MinIO systemd service file
+    create_enable_systemd_service
+    # add MinIO section to fstab
+    msg info "Add MinIO section to fstab"
+    cat >> /etc/fstab <<EOF
+# Start MinIO-volumes
+# End MinIO-volumes
+EOF
+    
+    # service metadata
+    create_one_service_metadata
+
+    # cleanup
+    postinstall_cleanup
+
+    msg info "INSTALLATION FINISHED"
+
+    return 0
+}
+
+service_configure()
+{
+    # For each drive other than /dev/vda, check if they're already formatted. 
+    # If there's no format, create partitions and filesystem
+    msg info "Checking drives partition and format"
+    minio_drives=$(lsblk -d -o NAME | grep vd | grep -v 'vda')
+    drives_count=1
+    MINIO_VOLUMES=""
+    for device in $minio_drives; do
+        # Partition and format dev
+        partition_format_drive $device
+        # Create folder in /mnt for drive mount
+        folder="/mnt/disk-${drives_count}"
+        MINIO_VOLUMES+=" ${folder}"
+        msg info "Create folder ${folder}"
+        if [[ $(ls $folder) ]]; then
+            mkdir $folder
+            msg info "Add /dev/${device}1 entry to fstab"
+            sed -i "/# End MinIO-volumes/i /dev/${device}1    ${folder}    xfs    defaults    0    0" /etc/fstab
+        else
+            msg info "Folder ${folder} exists, skipping"
+        fi
+        
+        drives_count=$((drives_count+1))
+    done
+    drives_count=$((drives_count-1))
+
+    msg info "Mount fstab"
+    [[ ! $(mount -a -t xfs) ]] || msg error "Error mounting MinIO drives"
```

Comment:
This is a weird construction, simple if statemant would be better, something like:

```shell
if ! mount -a -t xfs; then
    msg error "Error mounting MinIO drives"
    exit 1
fi  
```